R/board_s3.R

```diff
@@ -221,6 +221,34 @@ pin_store.pins_board_s3 <- function(board, name, paths, metadata,
   name
 }
 
+#' @rdname board_deparse
+#' @export
+board_deparse.pins_board_s3 <- function(board, ...) {
+  bucket <- check_board_deparse(board, "bucket")
+  if ("prefix" %in% names(board)) {
+    prefix <- board[["prefix"]]
+  } else {
+    prefix <- NULL
+  }
+
+  ## non-auth config
+  config <- board$svc$.internal$config
+  region <- config$region
+  if (nchar(region) == 0) region <- NULL
+  endpoint <- config$endpoint
+  if (nchar(endpoint) == 0) endpoint <- NULL
+  profile <- config$credentials$profile
+  if (nchar(profile) == 0) profile <- NULL
```

Comment:
I don't have quite enough experience with AWS to know if `profile` is more like server-level config or more like auth. Should it stay with the board or should we expect people to have it on their remote server in something like an environment variable?